sdks/rust/src/db_connection.rs

```diff
@@ -908,6 +961,58 @@ but you must call one of them, or else the connection will never progress.
         Ok(ctx_imp)
     }
 
+    #[cfg(feature = "web")]
+    pub async fn build_impl(self) -> crate::Result<DbContextImpl<M>> {
+        let db_callbacks = DbCallbacks::default();
+        let reducer_callbacks = ReducerCallbacks::default();
+
+        let connection_id_override = get_connection_id_override();
+        let ws_connection = WsConnection::connect(
+            self.uri.unwrap(),
+            self.module_name.as_ref().unwrap(),
+            self.token.as_deref(),
+            connection_id_override,
+            self.params,
+        )
+        .await
+        .map_err(|source| crate::Error::FailedToConnect {
+            source: InternalError::new("Failed to initiate WebSocket connection").with_cause(source),
+        })?;
+
+        let (raw_msg_recv, raw_msg_send) = ws_connection.spawn_message_loop();
+        let parsed_recv_chan = spawn_parse_loop::<M>(raw_msg_recv);
+
+        let inner = Arc::new(StdMutex::new(DbContextImplInner {
+            db_callbacks,
+            reducer_callbacks,
+            subscriptions: SubscriptionManager::default(),
+
+            on_connect: self.on_connect,
+            on_connect_error: self.on_connect_error,
+            on_disconnect: self.on_disconnect,
+            call_reducer_flags: <_>::default(),
+        }));
+
+        let mut cache = ClientCache::default();
+        M::register_tables(&mut cache);
+        let cache = Arc::new(StdMutex::new(cache));
+        let send_chan = Arc::new(StdMutex::new(Some(raw_msg_send)));
+
+        let (pending_mutations_send, pending_mutations_recv) = mpsc::unbounded();
+        let ctx_imp = DbContextImpl {
+            inner,
+            send_chan,
+            cache,
+            recv: Arc::new(StdMutex::new(parsed_recv_chan)),
+            pending_mutations_send,
+            pending_mutations_recv: Arc::new(StdMutex::new(pending_mutations_recv)),
+            identity: Arc::new(StdMutex::new(None)),
+            connection_id: Arc::new(StdMutex::new(connection_id_override)),
+        };
+
+        Ok(ctx_imp)
+    }
```

Comment:
Please move shared parts of this and the non-web `build_impl` into smaller functions/methods to reduce duplication.